File: samples/android-application/src/main/java/sample/android/application/MainActivity.java
package sample.android.application;

import android.app.Activity;
import android.os.Bundle;
import android.widget.TextView;

public final class MainActivity extends Activity {
    public static final String DISPLAY_CONSTANT = "android-application-static-final-value";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        TextView message = new TextView(this);
        message.setText(complexValue("android-application-beta-key"));
        setContentView(message);
    }

    public String complexValue(String key) {
        String selected;
        switch (key) {
            case "android-application-alpha-key":
                selected = "android-application-switch-alpha";
                break;
            case "android-application-beta-key":
                selected = "android-application-switch-beta\twith-tab";
                break;
            default:
                selected = "android-application-switch-default";
        }
        StringSupplier lambda = () -> "android-application-lambda-value";
        return DISPLAY_CONSTANT
               + "|android-application-line-one\nandroid-application-line-two"
               + "|android-application-unicode-\u4F60\u597D-\uD83D\uDE80|"
               + selected
               + "|"
               + lambda.get();
    }

    private interface StringSupplier {
        String get();
    }
}

File: samples/kotlin-android-application/src/main/kotlin/sample/kotlin/android/MainActivity.kt
package sample.kotlin.android

import android.app.Activity
import android.os.Bundle
import android.widget.TextView

class MainActivity : Activity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(TextView(this).apply { text = "kotlin-android-protected-value" })
    }

    fun identityFirst(): String = "kotlin-android-shared-identity"

    fun identitySecond(): String = "kotlin-android-shared-identity"

    fun specialUtf16(): String = "prefix\u0000\uD800middle\uDC00\uD83D\uDE00suffix"

    fun arrayValue(index: Int): String =
        arrayOf(
            "kotlin-android-array-zero",
            "kotlin-android-line-one\nkotlin-android-line-two",
            "kotlin-android-unicode-\u4F60\u597D-\uD83D\uDE80",
        )[index]

    fun whenValue(code: Int): String =
        when (code) {
            0 -> "kotlin-android-when-zero"
            1 -> "kotlin-android-when-one\twith-tab"
            else -> "kotlin-android-when-default"
        }

    fun lambdaValue(): String = { "kotlin-android-lambda-value" }()
}

object AndroidIdentityPeer {
    fun sharedIdentity(): String = "kotlin-android-shared-identity"

    fun concurrentIdentity(): String = "kotlin-android-concurrent-first-access"
}
